src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs

```diff
@@ -684,7 +686,31 @@ private async Task Orphan(IHost instance, CancellationToken cancellationToken =
                 if (instance != null)
                 {
                     GetHostLogger(instance).LogDebug("Disposing ScriptHost.");
-                    instance.Dispose();
+
+                    if (isStandbyHost && !_scriptWebHostEnvironment.InStandbyMode)
+                    {
+                        // For cold start reasons delay disposing script host if specializing out of placeholder mode
+                        Utility.ExecuteAfterColdStartDelay(_environment, () =>
+                        {
+                            try
+                            {
+                                GetHostLogger(instance).LogDebug("Starting Standby ScriptHost dispose");
+                                instance.Dispose();
+                                _logger.LogDebug("Standby ScriptHost disposed");
+                            }
+                            catch (Exception e)
+                            {
+                                _logger.LogError(e, "Failed to dispose Standby ScriptHost instance");
```

Comment:
is try...catch needed? Ok to keep running even this disposal fails?

Reply:
i think it's okay to catch and log -- but we should re-throw the exception to match the previous behavior (of not catching).

Reply:
Thanks. Yes didnt mean to change that behavior.

Reply:
yes. didn't mean to change that behavior. Fixed.